### PR TITLE
Add kestrel tests.

### PR DESCRIPTION
**Pull request by Sridhar-MS**
Tests will 'build', 'run', 'publish' and 'execute' a Kestrel Hello World server as a PortableFatApp and as a Standalone app.

cc @piotrpMSFT @brthor @livarcocc @krwq

---
This change is [Reviewable](https://reviewable.io/reviews/dotnet/cli/2269)

### CODE REVIEW

**Comment by muratg** [comment]
:eyes:

File: TestAssets/TestProjects/KestrelSample/KestrelPortable/project.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
+  "version": "1.0.0-*",
+  "dependencies": {    
+    "Microsoft.AspNetCore.Server.Kestrel": "1.0.0-rc2-20113",
+    "Microsoft.AspNet.Hosting": "1.0.0-rc2-16253",
```
Comment (BrennanConroy) [inline TestAssets/TestProjects/KestrelSample/KestrelPortable/project.json]: This is a super old package, Microsoft.AspNetCore.Hosting

Comment (BrennanConroy) [inline TestAssets/TestProjects/KestrelSample/KestrelPortable/project.json]: This is a super old package, Microsoft.AspNetCore.Hosting

**Comment by Sridhar-MS** [comment]
Github and jenkins CI has been behaving strangely. Only 4 check are run.

**Comment by Sridhar-MS** [comment]
@dotnet-bot test this please

**Comment by muratg** [comment]
As @BrennanConroy pointed out, please do not use `AspNet.*` packages. Only `AspNetCore.*` packages!

**Comment by muratg** [comment]
@Sridhar-MS BTW, thanks for this change! Hopefully we'll catch more issues earlier.

Any chance you could also add HelloMvc?  It has more coverage.

File: TestAssets/TestProjects/KestrelSample/KestrelPortable/project.json (hunk 1 above)
Comment (Sridhar-MS) [inline TestAssets/TestProjects/KestrelSample/KestrelPortable/project.json]: Fixed it

**Comment by Sridhar-MS** [comment]
@muratg I have created issue #2303 for adding HelloMvc.

File: test/Kestrel.Tests/DotnetTest.cs [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+                NetworkHelper.IsServerUp(url).Should().BeTrue($"Unable to connect to kestrel server - {KestrelStandalone} @ {url}");
+                NetworkHelper.TestGetRequest(url, args);
+            }
+            finally
+            {
+                command.KillTree();
+            }
+        }
+
+        private static string Build(string testRoot)
```
Comment (brthor) [inline test/Kestrel.Tests/DotnetTest.cs]: I've seen this function in a few of the test classes. We should add this to the test framework in the future.

File: test/Microsoft.DotNet.Tools.Tests.Utilities/Commands/DotnetCommand.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using Microsoft.DotNet.Cli.Utils;
+
+namespace Microsoft.DotNet.Tools.Test.Utilities
+{
+    public sealed class DotnetCommand : TestCommand
```
Comment (brthor) [inline test/Microsoft.DotNet.Tools.Tests.Utilities/Commands/DotnetCommand.cs]: What does this give us? A muxer invocation?

File: test/Microsoft.DotNet.Tools.Tests.Utilities/Commands/RunCommand.cs [hunk 4]
```diff
@@ -60,10 +61,10 @@ private string AppArgsArgument
 
         public RunCommand(
             string projectPath,
-            string framework="",
-            string configuration="",
-            bool preserveTemporary=false,
-            string appArgs="")
+            string framework = "",
```
Comment (brthor) [inline test/Microsoft.DotNet.Tools.Tests.Utilities/Commands/RunCommand.cs]: :smile: python style -> c# style

File: test/Microsoft.DotNet.Tools.Tests.Utilities/Commands/TestCommand.cs [hunk 5]
```diff
@@ -15,6 +17,8 @@ public class TestCommand
 
         public string WorkingDirectory { get; set; }
 
+        public Process CurrentProcess { get; set; }
```
Comment (brthor) [inline test/Microsoft.DotNet.Tools.Tests.Utilities/Commands/TestCommand.cs]: I don't feel that this is right that the TestCommand class holds the state of a running process. What if we wanted to execute two of the same to test concurrency issues?

File: test/Microsoft.DotNet.Tools.Tests.Utilities/Commands/TestCommand.cs [hunk 6]
```diff
@@ -55,7 +75,17 @@ public virtual CommandResult ExecuteWithCapturedOutput(string args = "")
 
             return RunProcess(commandPath, args, stdOut, stdErr);
         }
-        
+
+        public void KillTree()
```
Comment (brthor) [inline test/Microsoft.DotNet.Tools.Tests.Utilities/Commands/TestCommand.cs]: Same here, why can't RunAsync return the process?

File: test/Microsoft.DotNet.Tools.Tests.Utilities/ProcessExtensions.cs [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+    {
+        private static readonly bool _isWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+        private static readonly TimeSpan _defaultTimeout = TimeSpan.FromSeconds(30);
+
+        public static void KillTree(this Process process)
+        {
+            process.KillTree(_defaultTimeout);
+        }
+
+        public static void KillTree(this Process process, TimeSpan timeout)
```
Comment (brthor) [inline test/Microsoft.DotNet.Tools.Tests.Utilities/ProcessExtensions.cs]: This is pretty cool

**Comment by brthor** [comment]
Made a few comments but don't block on them :shipit:

File: test/Microsoft.DotNet.Tools.Tests.Utilities/Commands/DotnetCommand.cs (hunk 3 above)
Comment (Sridhar-MS) [inline test/Microsoft.DotNet.Tools.Tests.Utilities/Commands/DotnetCommand.cs]: Yup.

File: test/Microsoft.DotNet.Tools.Tests.Utilities/Commands/RunCommand.cs (hunk 4 above)
Comment (Sridhar-MS) [inline test/Microsoft.DotNet.Tools.Tests.Utilities/Commands/RunCommand.cs]: This is just Visual Studio magic Ctrl K + Ctrl D

**Comment by Sridhar-MS** [comment]
@brthor  filed #2318

**Pull request merged**